src/client/dfuse/dfuse_main.c

```diff
@@ -364,6 +371,26 @@ main(int argc, char **argv)
 		D_GOTO(out_debug, ret = -DER_NO_HDL);
 	}
 
+	if (dfuse_info->di_threaded && !have_thread_count) {
+		cpu_set_t cpuset;
+
+		rc = sched_getaffinity(0, sizeof(cpuset), &cpuset);
+		if (rc != 0) {
+			printf("Failed to get cpuset information\n");
+			exit(1);
+		}
+
+		dfuse_info->di_thread_count = CPU_COUNT(&cpuset);
+
+		/* Reserve one CPU thread for the daos event queue */
+		dfuse_info->di_thread_count -= 1;
+
+		if (dfuse_info->di_thread_count < 1) {
+			printf("Dfuse needs more threads\n");
```

Comment:
print this to stderr?
probably some more info would help (di_thread_count - 1)

Reply:
I'd fix the output in a future PR.  I'm adding new command line options and help for login-node support anyway.